services/MerchStoreService.ts

```diff
@@ -100,47 +100,70 @@ export default class MerchStoreService {
     const item = await this.transactions.readOnly(async (txn) => Repositories
       .merchStoreItem(txn)
       .findByUuid(uuid));
-    if (!item) throw new NotFoundError('Item not found');
+    if (!item) throw new NotFoundError('Merch item not found');
     return item.getPublicMerchItem();
   }
 
   public async createItem(item: MerchItem): Promise<MerchandiseItemModel> {
     return this.transactions.readWrite(async (txn) => {
+      item.hasVariantsEnabled ??= false;
```

Comment:
An assignment like this belongs in the controller I think, since it's request-related ("did the request send in a value for `hasVariantsEnabled`").